### DISCUSSION TITLE
对评论框网址没添加"https"的进行自动添加获取改进？

### DISCUSSION THREAD

**Discussion by xiamuguizhi** [ArtalkJS/Artalk#129] [Ideas]
有的访客评论有时候网址 不会带 ` https:// ` 就单纯留个博客地址 如： `biji.io `  ，这也导致了 如果点击链接访问 这个用户博客就变成了 `https://xn--qpru0x.cn/post/biji.io` 这样了。

![image](https://user-images.githubusercontent.com/19650297/165256687-42a5e660-63ef-49d0-9910-0b1b31b3b0a0.png)
 
是否可以在提交评论后 判断是否 存在 `https://` 自行添加 或者 在前端输出的时候 判断添加个  `https://`

**Comment by qwqcode** [comment]
这个 bug 确实存在！

**Comment by qwqcode** [comment]
![image](https://user-images.githubusercontent.com/22412567/165257874-b86a6d91-6c3c-46ee-857e-95b382b1eeae.png)

![image](https://user-images.githubusercontent.com/22412567/165257901-ce2fa438-4d00-454f-8fd4-c29ca9b9fbcd.png)

后端合法性判断出了问题，应该是 `IsRequestURL` 而不是 `IsURL` 🤣🤣🤣🤣

**Comment by qwqcode** [comment]
我会在前端加判断，如果 URL 的 协议不存在时，自动加上 https

**Reply by xiamuguizhi** [reply]
非常感谢 ，期待修复

**Reply by qwqcode** [reply]
后端已经修了 https://github.com/ArtalkJS/ArtalkGo/commit/2ad686433a804bade1be59aceafb7c4e89395721 正在编译

**Reply by xiamuguizhi** [reply]
@qwqcode 我刚更新好 `Artalk (v2.1.4/2ad6864)` 但是我想问一下，它是对以后这样的问题修复是吗？因为我发现之前的 还是这样 `https://xn--qpru0x.cn/post/biji.io`  。

我以为对以前的也有效果 哈哈。

**Reply by qwqcode** [reply]
以前的可以 users 数据表里面改改，然后刷新下 ArtalkGo 的缓存

![image](https://user-images.githubusercontent.com/22412567/165264538-80eb3b4b-9951-49a7-830f-c6099785d2ae.png)

**Reply by qwqcode** [reply]
前端刚刚也修了，但是还没发布新版，当没有输入带 http 或 https 的连接时，会自动补全

https://github.com/ArtalkJS/Artalk/commit/667fc66397d022bcea8ab679a63d752cb72ea1e2

**Reply by xiamuguizhi** [reply]
@qwqcode 又更新到了 `Artalk (v2.1.4/2ad6864)`  通过数据库添加了 `https`  在清除了缓存 ， 现在一切正常  感谢！
